integration_tests/src/examples.rs

```diff
@@ -299,3 +468,141 @@ fn get_formated_body(body_str: &str) -> String {
         msg_string,
     )
 }
+
+pub enum TestSample {
+    BigBuckBunny,
+    ElephantsDream,
+    Sample,
+    SampleLoop,
+    Generic,
+}
+
+pub enum TestCodec {
+    AAC,
+    LIBOPUS,
+}
+
+pub fn ff_stream_sample(
+    ip: &str,
+    video_port: Option<u16>,
+    audio_port: Option<u16>,
+    test_sample: TestSample,
+) -> Result<()> {
+    match test_sample {
+        TestSample::BigBuckBunny => ff_stream(
+            ip,
+            video_port,
+            audio_port,
+            get_asset_path(test_sample).unwrap(),
+        ),
+        TestSample::ElephantsDream => ff_stream(
+            ip,
+            video_port,
+            audio_port,
+            get_asset_path(test_sample).unwrap(),
+        ),
+        TestSample::Sample => {
+            if let Some(port) = video_port {
+                ff_stream_video(ip, port, get_asset_path(test_sample).unwrap())
+            } else {
+                Err(anyhow!("video port required for test sample"))
+            }
+        }
+        TestSample::SampleLoop => {
+            if let Some(port) = video_port {
+                ff_stream_video_loop(ip, port, get_asset_path(test_sample).unwrap())
+            } else {
+                Err(anyhow!("video port required for test sample"))
+            }
+        }
+        TestSample::Generic => {
+            if let Some(port) = video_port {
+                ff_stream_testsrc(
+                    ip,
+                    port,
+                    Resolution {
+                        width: 1920,
+                        height: 1080,
+                    },
+                )
+            } else {
+                Err(anyhow!("video port required for generic"))
+            }
+        }
+    }
+}
+
+// receives and plays h264 encoded video and opus encoded audio
+pub fn gst_stream_sample(
+    ip: &str,
+    video_port: Option<u16>,
+    audio_port: Option<u16>,
+    test_sample: TestSample,
+) -> Result<()> {
+    match test_sample {
+        TestSample::BigBuckBunny => gst_stream(
+            ip,
+            video_port,
+            audio_port,
+            path_to_str(get_asset_path(test_sample).unwrap()).as_str(),
+        ),
+        TestSample::ElephantsDream => gst_stream(
+            ip,
+            video_port,
+            audio_port,
+            path_to_str(get_asset_path(test_sample).unwrap()).as_str(),
+        ),
+        TestSample::Sample => gst_stream(
+            ip,
+            video_port,
+            audio_port,
+            path_to_str(get_asset_path(test_sample).unwrap()).as_str(),
+        ),
+        TestSample::SampleLoop => Err(anyhow!(
+            "cannot play sample in loop using gstreamer, try ffmpeg"
+        )),
+        TestSample::Generic => gst_stream_testsrc(ip, video_port, audio_port),
+    }
+}
+
+fn path_to_str(path_buf: PathBuf) -> String {
```

Comment:
I would avoid helpers like this, maybe do the conversion to string inside `gst_stream`?